app/scripts/views/mixins/connect-another-device-mixin.js

```diff
@@ -0,0 +1,146 @@
+      if (! this.isEligibleForConnectAnotherDevice(account)) {
+        // this shouldn't happen IRL.
+        return p.reject(new Error('chooseConnectAnotherDeviceScreen can only be called if user is eligible to connect another device'));
+      }
+
+      return this._isEligibleForSms(account)
+        .then(({ ok, country }) => {
+          if (ok) {
+            // User is eligible for SMS experiment, now bucket
+            // users into treatment and control groups.
+            const group = this.getExperimentGroup('sendSms', { account });
+            this.createExperiment('sendSms', group);
+
+            if (group === 'treatment') {
+              this.navigate('sms', { account, country });
+            } else { // there are only two groups, by default, this is the control
+              this.navigate('connect_another_device', { account });
+            }
+          } else {
+            this.navigate('connect_another_device', { account });
+          }
+        });
+    },
+
+    /**
+     * Check if the requirements are met to send an SMS.
+     *
+     * @param {Object} account
+     * @returns {Boolean}
+     * @private
+     */
+    _areSmsRequirementsMet (account) {
+      return this.isSignUp() &&
+        // If already on a mobile device, doesn't make sense to send an SMS.
+        ! this.getUserAgent().isAndroid() &&
+        ! this.getUserAgent().isIos() &&
+        // If a user is already signed in to Sync which is different to the
+        // user that just verified, show them the old "Account verified!" screen.
+        ! this._isAnotherUserSignedIn(account) &&
+        // Does able say we are eligible for the experiment?
+        this.isInExperiment('sendSms', { account });
+    },
+
+    /**
+     * Is `account` eligible for SMS? Does not check country.
+     *
+     * @param {Object} account
+     * @returns {Promise} resolves to an object with two fields:
+     *   @returns {String} country - country user is in, only valid if user is eligible for SMS
+     *   @returns {Boolean} ok - whether the user is eligible for SMS.
+     * @private
+     */
+    _isEligibleForSms (account) {
+      return p(
+        this._areSmsRequirementsMet(account) &&
+        this._smsCountry(account)
+      )
+      .then((country) => {
+        return {
+          country,
+          ok: !! country
+        };
+      });
+    },
+
+    /**
+     * Check if `account` to send an SMS, and if so, which
+     * country the SMS should be sent to.
+     *
+     * @param {Object} account - account to check
+     * @returns {Promise} If user can send an SMS, resolves to
+     *   the country to send the SMS to.
+     * @private
+     */
+    _smsCountry (account) {
+      // The auth server can gate whether users can send an SMS based
+      // on the user's country and whether the SMS provider account
+      // has sufficient funds.
+      return account.smsStatus(this.relier.pick('country'))
```

Comment:
Something about relying on `this.relier` here bothers me. I think because it's potentially unexpected. Maybe `country` should be passed in...